### PR TITLE
Fixed editor throwing an exception when destroyed shortly after it was created

### LINKED ISSUE #2257: Editor throws an exception when destroyed shortly after it was created
## Type of report

Bug

## Provide detailed reproduction steps (if any)

There's a possible race condition if you delete the editor right after it was instantiated.

1. Open [codepen demo](https://codepen.io/mlewand/pen/wxejOg).
1. Open JavaScript console.
1. Click the "Load/Destory" button.

### Expected

Nothing happens.

## Actual

`Uncaught TypeError: Cannot read property 'allow' of undefined` exception is being thrown.

Note that it could sometimes require like 3 or 5 attempts to repro.

## Other details

Originaly reported in old tracker as [#14613](https://dev.ckeditor.com/ticket/14613).

* Browser: Any
* OS: Any

### PR DESCRIPTION
**Pull request by mlewand**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests) and
[how to create tests](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [ ] Manual tests

Skipped the manual test as this is pretty much an internal change, and the case presented for manually reproduced the issue required refreshing the page.

## What changes did you make?

This PR adds two early returns, making sure that delayed editor initialization doesn't take place in case it was removed before.

4 people in total worked on this one :D I had to adjust the unit tests, as they were not failing for previous CKEditor releases, thus it would not protect us from a regression.

This is a follow-up of a #273 PR.

Closes #2257, #718.

### CODE REVIEW

File: tests/plugins/wysiwygarea/destroy.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/* bender-tags: editor,unit */
+/* bender-ckeditor-plugins: toolbar */
+
+( function() {
+	'use strict';
+
+	bender.test( {
+		// (#2257)
+		'test destroy editor on instance created': function() {
```
Comment (jacekbogdanski) [inline tests/plugins/wysiwygarea/destroy.js]: The test always passes despite changes.

**Comment by mlewand** [comment]
Fixed the unit test.

I decided to **remove** `wysiwygarea` change, as it does not fix the issue, there's still an option to break it with the following code: https://gist.github.com/mlewand/ee09f6dc0148792c8d08f53cc55da605 However fixing it would take more time (it would need to be done very carefully in core setMode logic) and it's just an edge case.

As for manual test I related to that in:

> Skipped the manual test as this is pretty much an internal change, and the case presented for manually reproduced the issue required refreshing the page.

**Comment by f1ames** [comment]
Closing this PR as there is already more general PR on the way https://github.com/ckeditor/ckeditor-dev/pull/3128.
